### DISCUSSION TITLE
Two forms share `useFormContext` unmounted component issue

### DISCUSSION THREAD

**Discussion by slamdunkYYY** [react-hook-form/react-hook-form#7300] [Q&A]
I have two forms, supposing formA and formB. I use one `useFormContext` as a provider and inside the provider I have that both forms. The two forms are in two routes. Both forms are working fine when on individual page. However, when I load one form on one page first and then switch the page via nav tab to load another page, after I submit that form, I receive the error message says `Can't perform a React state update on an unmounted component` issue.

In one form, I use `trigger` on `useEffect`. I'm not sure if this makes the issue...

If I have to use multiple forms but with one form context only, how should I fix this issue?

**Comment by bluebill1049** [comment]
form context should applied per form

**Reply by slamdunkYYY** [reply]
If I want to share errors between forms, should I put those errors on a state management?
